src/seedsigner/helpers/mnemonic_generation.py

```diff
@@ -37,28 +85,227 @@ def calculate_checksum(mnemonic: list, wordlist_language_code: str) -> List[str]
 
 
 
-def generate_mnemonic_from_dice(roll_data: str) -> List[str]:
+def generate_mnemonic_from_dice(roll_data: str, wordlist_language_code: str = WORDLIST_LANGUAGE__ENGLISH) -> list[str]:
+    """
+        Takes a string of 50 or 99 dice rolls and returns a 12- or 24-word mnemonic.
+
+        Uses the iancoleman.io/bip39 and bitcoiner.guide/seed "Base 10" or "Hex" mode approach:
+        * dice rolls are treated as string data.
+        * hashed via SHA256.
+
+        Important note: This method is NOT compatible with iancoleman's "Dice" mode.
+    """
     entropy_bytes = hashlib.sha256(roll_data.encode()).digest()
 
-    if len(roll_data) == 50:
+    if len(roll_data) == DICE__NUM_ROLLS__12WORD:
         # 12-word mnemonic; only use 128bits / 16 bytes
         entropy_bytes = entropy_bytes[:16]
 
     # Return as a list
-    return bip39.mnemonic_from_bytes(entropy_bytes).split()
+    return bip39.mnemonic_from_bytes(entropy_bytes, wordlist=_get_wordlist(wordlist_language_code)).split()
+
+
+
+def generate_mnemonic_from_coin_flips(coin_flips: str, wordlist_language_code: str = WORDLIST_LANGUAGE__ENGLISH) -> list[str]:
+    """
+        Takes a string of 128 or 256 0s and 1s and returns a 12- or 24-word mnemonic.
+
+        Uses the iancoleman.io/bip39 and bitcoiner.guide/seed "Binary" mode approach:
+        * binary digit stream is treated as string data.
+        * hashed via SHA256.
+    """
+    entropy_bytes = hashlib.sha256(coin_flips.encode()).digest()
+
+    if len(coin_flips) == 128:
+        # 12-word mnemonic; only use 128bits / 16 bytes
+        entropy_bytes = entropy_bytes[:16]
+
+    # Return as a list
+    return bip39.mnemonic_from_bytes(entropy_bytes, wordlist=_get_wordlist(wordlist_language_code)).split()
+
+
+
+def get_partial_final_word(coin_flips: str, wordlist_language_code: str = WORDLIST_LANGUAGE__ENGLISH) -> str:
+    """ Look up the partial final word for the given coin flips.
+        7 coin flips: 0101010 + 0000 where the final 4 bits will be replaced with the checksum
+        3 coin flips: 0101 + 0000000 where the final 8 bits will be replaced with the checksum
+    """
+    binary_string = coin_flips + "0" * (11 - len(coin_flips))
+    wordlist_index = int(binary_string, 2)
+
+    return _get_wordlist(wordlist_language_code)[wordlist_index]
 
 
 
 # Note: This currently isn't being used since we're now chaining hashed bytes for the
 #   image-based entropy and aren't just ingesting a single image.
-def generate_mnemonic_from_image(image) -> List[str]:
+def generate_mnemonic_from_image(image, wordlist_language_code: str = WORDLIST_LANGUAGE__ENGLISH) -> list[str]:
     import hashlib
     hash = hashlib.sha256(image.tobytes())
 
     # Return as a list
-    return bip39.mnemonic_from_bytes(hash.digest()).split()
+    return bip39.mnemonic_from_bytes(hash.digest(), wordlist=_get_wordlist(wordlist_language_code)).split()
+
+
+
+if __name__ == "__main__":
```

Comment:
Besides another sole instance in settings_definitions.py, nowhere else in our codebase are we already using this convention -- of an executable cli/library.  Would this entire `if __name__ == "__main__":` block be more at home in tools/???

I tried moving this "__main__" block into tools/mnemonic_generation.py.  It makes importing seedsigner a strange hack... but if user installed seedsigner locally, it would just be adding a `from seedsigner.helpers.mnemonic_generation import *` and then we wouldn't have cli code that exists in the standard codebase except where users clone the repo.

It's just a thought, I'm still not sure.  It's so common while prototyping, maybe less common in production unless users can normally use it (not the case for code in seedsigner-os).

Reply:
If the concern/goal is to keep the CLI code out of the standard release package, I think there are better ways to package it (e.g. its own dir outside of `src` that isn't included in a release or a dir in `src` that is explicitly excluded from the build process).

That being said, for now I like the directness of the organization here: this file does X for SeedSigner, run its CLI to play with X interactively yourself. No imports (aside from `embit`) or redirects or obfuscation. This code does X. Period.

Reply:
I hereby withdraw my comments above.

I too see the value in the way it is all-in-one.  Can verify dice rolls with the exact file that seedsigner is using.